### DISCUSSION TITLE
A research for generating PR checklists in Pull Request Template

### DISCUSSION THREAD

**Discussion by baishuotong** [localForage/localForage#1104] [Show and tell]
Dear maintainers and communities interested in this project:

Our team is currently developing a tool for automatically generating PR checklists in Pull Request Template. The tool generates a set of PR rules based on the contribution guidelines in your project, which external contributors may be expected to adhere to. Below are the results of our generation:

**Test**: 
1.Add test, if you add features or fix bug, please write or add test.
2.All code can be tested for style by running the unit tests (npm test). 

**style**:
This will make sure your code conforms to our styleguide in .jshint and .jscsrc

**Branch**:
1.Work on a feature branch
2.Try to rebase master into your branch.

**commit message**:
1.Run grunt build before you commit
2.All drivers must have the same public API
3.Please commit changes at the top-level folder along with your changes in the src/ folder
4.Not make these changes separate commits

Also, we suggest you add some checklist notes about, **Documentation**, **Title**, **Description**, **contributing** and **CLA** which are also important for submitting PRs according to our analysis.
We hope you and your team can provide feedback on our generated output.

Best regards.

**Comment by lxstart1024** [comment]
I might check contributing file. Even if some projects have checklists, I still read the contributing to prevent missing some details.
